Odata-docs/performance/disable-writer-validations.md

```diff
@@ -0,0 +1,137 @@
+
+During serialization, the OData writer performs various validations to ensure that the payload is valid (e.g. no duplicate properties) and conforms to the schema (e.g. property type matches the entity type's definition). These validations introduce some overhead to the serialization process. If you know ahead of time that the payload you're writing is correct, you can disable one or more validations to improve performance.
+
+Use the [`ODataMessageWriterSettings.Validations`](/dotnet/api/microsoft.odata.odatamessagewritersettings.validations) property to configure which validations you want to enable or disable.
+
+You can disable all validations using [`ValidationKinds.None`](/dotnet/api/microsoft.odata.validationkinds):
+
+```csharp
+messageWriterSettings.Validations = ValidationKinds.None;
+```
+
+You can selectively enable validations using the `&` operator:
+
+```csharp
+messageWriterSettings.Validations = ValidationKinds.ThrowOnDuplicatePropertyNames & ValidationKinds.ThrowIfTypeConflictsWithMetadata
+```
+
+You can enable all validations using [`ValidationKinds.All`](/dotnet/api/microsoft.odata.validationkinds):
+
+```csharp
+messageWriterSettings.Validations = ValidationKinds.All;
+```
+
+To learn more about different validation options, consult the [`ValidationKinds` enum documentation](/dotnet/api/microsoft.odata.validationkinds).
+
+## Disabling writer validations when using `Microsoft.OData.Core` library directly
+
+When using the core library directly, you pass the [`ODataMessageWriterSettings`](/dotnet/api/microsoft.odata.odatamessagewritersettings) to the [`ODataMessageWriter`](/dotnet/api/microsoft.odata.odatamessagewriter) construtor.
+
+```csharp
+ODataMessageWriterSettings settings = new ODataMessageWriterSettings
+{
+    ODataUri = new ODataUri() { /* ... */ },
+    Validations = ValidationKinds.None
+};
+
+ODataMessageWriter writer = new ODataMessageWriter(odataMessage, settings, edmModel);
+
+// ...
+```
+
+## Disabling writer validations when using `Microsoft.AspNetCore.OData` 8 library
+
+In `Microsoft.AspNetCore.OData` 8, you can pass a service configuration action as the third argument of `AddRouteComponents`. Use
+this to inject a request-scoped instance of `ODataMessageWriterSettings` - with the desired options - to the service container:
+
+```csharp
+services
+.AddControllers()
+.AddOData(options =>
+    options.AddRouteComponents(
+        routePrefix: "routePrefix",
+        model: edmModel,
+        configurateServices: container =>
+        {
+            container.AddScoped<ODataMessageWriterSettings>(_ => new ODataMessageWriterSettings
+            {
+                Validations = ValidationKinds.None
+            })
+        }));
+```
+
+## Disabling writer validations when using `Microsoft.AspNetCore.OData` 7 library
```

Comment:
Is this `Microsoft.AspNetCore 7 only or even Microsoft.AspNet.OData`

Reply:
Yes. `ODataUtf8JsonWriter` is not supported in `Microsoft.AspNet.OData` because it depends on `Utf8JsonWriter` which is not supported in .NET Framework.

Reply:
@habbes I don't think this particular article is related to `ODataUtf8JsonWriter`, is it?

Reply:
Hmm. You are correct. I had confused this with a different conversation. Let me revert back with a response.

Reply:
You can do this in AspNet.OData 7 as well using the `MapODataServiceRoute` method like the MVC routing approach below. The only difference is that the AspNet 7.x configuration setup looks different from the AspNetCore one (there's no Startup class with a Configure method, there's a `WebApiConfig` class with a `Register` method that takes a `HttpConfiguration` argument.

But the OData part is identical. I think readers would still be able to easily adapt this sample to their AspNet.OData code.